### PR TITLE
fix: ignore body should not set content-length 

### LINKED ISSUE #320: fix: ignore body should not set content-length
**Describe the bug**

refer to fasthttp https://github.com/valyala/fasthttp/pull/1019 and https://github.com/valyala/fasthttp/pull/1022

### PR DESCRIPTION
**Pull request by byene0923**
#### What type of PR is this?
fix

#### Check the PR title.
- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: can refer to https://github.com/valyala/fasthttp/pull/1022, fasthttp only handle ContinueReadBody(), this pr also handle ContinueReadBodyStream() 
zh(optional):  fasthttp 只处理了 ContinueReadBody(), 这里添加了对 ContinueReadBodyStream() 的处理 和 测试

#### Which issue(s) this PR fixes:
close #320

### CODE REVIEW

File: pkg/protocol/http1/req/request_test.go [hunk 1]
```diff
@@ -92,6 +92,82 @@ func TestRequestContinueReadBody(t *testing.T) {
...
+	s := "POST / HTTP/1.1\r\n\r\n"
+
+	zr := mock.NewZeroCopyReader(s)
+	if err := Read(&r, zr); err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	r.SetHost("foobar")
+	headerStr := r.Header.String()
+	if !strings.Contains(headerStr, "Content-Length: ") {
+		t.Fatalf("unexpected Content-Length")
```
Comment (welkeyever) [inline pkg/protocol/http1/req/request_test.go:127]: 这个报错信息应该是should contains `Content-Length`?
另外可以把具体的值也解出来check下吧

File: pkg/protocol/http1/req/request_test.go [hunk 2]
```diff
@@ -92,6 +92,82 @@ func TestRequestContinueReadBody(t *testing.T) {
...
+	s := ""
+
+	zr := mock.NewZeroCopyReader(s)
+	if err := ContinueReadBodyStream(&r, zr, 2048, true); err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	r.SetHost("foobar")
+	headerStr := r.Header.String()
+	if strings.Contains(headerStr, "Content-Length: ") {
+		t.Fatalf("unexpected Content-Length")
```
Comment (welkeyever) [inline pkg/protocol/http1/req/request_test.go:147]: 同上

File: pkg/protocol/http1/req/request_test.go (hunk 1 above)
Comment (byene0923) [reply pkg/protocol/http1/req/request_test.go]: done

File: pkg/protocol/http1/req/request_test.go (hunk 2 above)
Comment (byene0923) [reply pkg/protocol/http1/req/request_test.go:151]: 这个可能评论错了，在TestRequestReadStreaming这里，我一并修改了

**Review by welkeyever — APPROVED** [review]

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**